tests/ttnn/unit_tests/gtests/test_graph_add.cpp

```diff
@@ -0,0 +1,204 @@
+        }
+
+        peak_memory_usage = std::max(peak_memory_usage, total_cb + total_buffer);
+    }
+
+    return peak_memory_usage;
+}
+
+// Returns count of intermediate and output tensors
+std::pair<int, int> count_intermediate_and_output_tensors(const nlohmann::json& trace) {
+    bool first_begin_found = false;
+    bool last_end_found = false;
+
+    std::unordered_set<int> intermediate_tensors;
+    std::unordered_set<int> output_tensors;
+
+    int first_begin_index = -1;
+    int last_end_index = -1;
+
+    for (int i = 0; i < trace.size(); ++i) {
+        const auto& v = trace[i];
+        if (v["name"] == "begin_function" && !first_begin_found) {
+            first_begin_found = true;
+            first_begin_index = i;
+        } else if (v["name"] == "end_function") {
+            last_end_found = true;
+            last_end_index = i;
+
+            if("create_device_tensor") {
+                auto id = v["connections"][0].get<int>();
+                intermediate_tensors.insert(id);
+            }
+        }
+    }
+
+    TT_ASSERT(first_begin_found);
+    TT_ASSERT(last_end_found);
+
+    for(int index : trace[last_end_index]["connections"]) {
+        output_tensors.insert(index);
+    }
+
+    for(int index : output_tensors) {
+        intermediate_tensors.erase(index);
+    }
+
+    // Return the counts of intermediate and output tensors
+    return {static_cast<int>(intermediate_tensors.size()), static_cast<int>(output_tensors.size())};
+}
+
+std::vector<std::string> extract_calltrace(const nlohmann::json& trace){
+    std::vector<std::string> op_calls;
+    size_t i = 0;
+
+    while (i < trace.size()) {
+        const auto& v = trace[i];
+        i++;
+
+        if (v["name"] == "begin_function") {
+            op_calls.push_back(v["params"]["name"]);
+        }
+    }
+
+    return op_calls;
+}
+}
+
+struct AddOpGraphTestParam {
+    ttnn::Shape a_Shape;
+    ttnn::Shape b_Shape;
+    ttnn::MemoryConfig memory_config; //DRAM_MEMORY_CONFIG, L1_MEMORY_CONFIG, L1_BLOCK_SHARDED_MEMORY_CONFIG, L1_HEIGHT_SHARDED_MEMORY_CONFIG, L1_WIDTH_SHARDED_MEMORY_CONFIG
+    std::vector<std::string> expected_calltrace;
+    uint32_t expected_peak_memory_usage = 0;
+    uint32_t expected_intermediate_tensors_count = 0;
+    uint32_t expected_output_tensors_count = 0;
+};
+
+class AddOpGraphTestFixture : public TTNNFixtureWithDevice,
+                              public testing::WithParamInterface<AddOpGraphTestParam> {};
+
+
+TEST_P(AddOpGraphTestFixture, AddGraphTrace) {
+    auto params = GetParam();
+    const auto device_id = 0;
+
+    {
+        const auto input_tensor_a = ttnn::zeros(params.a_Shape, ttnn::bfloat16, ttnn::TILE_LAYOUT, this->getDevice(), params.memory_config);
+        const auto input_tensor_b = ttnn::zeros(params.b_Shape, ttnn::bfloat16, ttnn::TILE_LAYOUT, this->getDevice(), params.memory_config);
+
+        ttnn::GraphProcessor::begin_graph_capture(tt::tt_metal::IGraphProcessor::RunMode::FAKE);
```

Comment:
please update to support a new name for RunMode and LGTM

Reply:
Done